packages/@aws-cdk/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -401,6 +401,130 @@ export enum FlowLogMaxAggregationInterval {
+
+  /**
+   * The destination port of the traffic.
+   */
+  public static readonly DST_PORT = LogFormatField.custom('dstport');
+
+  /**
+   * The IANA protocol number of the traffic.
+   */
+  public static readonly PROTOCOL = new LogFormatField('${protocol}');
+
+  /**
+   * The number of packets transferred during the flow.
+   */
+  public static readonly PACKETS = new LogFormatField('${packets}');
+
+  /**
+   * The number of bytes transferred during the flow.
+   */
+  public static readonly BYTES = new LogFormatField('${bytes}');
+
+  /**
+   * The packet-level (original) source IP address of the traffic.
+   */
+  public static readonly PKT_SRC_ADDR = new LogFormatField('${pkt-srcaddr}');
+
+  /**
+   * The packet-level (original) destination IP address for the traffic.
+   */
+  public static readonly PKT_DST_ADDR = new LogFormatField('${pkt-dstaddr}');
+
+  /**
+   * The Region that contains the network interface for which traffic is recorded.
+   */
+  public static readonly REGION = new LogFormatField('${region}');
+
+  /**
+   * The ID of the Availability Zone that contains the network interface for which traffic is recorded.
+   */
+  public static readonly AZ_ID = new LogFormatField('${az-id}');
+
+  /**
+   * The type of sublocation that's returned in the sublocation-id field.
+   */
+  public static readonly SUBLOCATION_TYPE = new LogFormatField('${sublocation-type}');
+
+  /**
+   * The ID of the sublocation that contains the network interface for which traffic is recorded.
+   */
+  public static readonly SUBLOCATION_ID = new LogFormatField('${sublocation-id}');
+
+  /**
+   * The name of the subset of IP address ranges for the pkt-srcaddr field,
+   * if the source IP address is for an AWS service.
+   */
+  public static readonly PKT_SRC_AWS_SERVICE = new LogFormatField('${pkt-src-aws-service}');
+
+  /**
+   * The name of the subset of IP address ranges for the pkt-dstaddr field,
+   * if the destination IP address is for an AWS service.
+   */
+  public static readonly PKT_DST_AWS_SERVICE = new LogFormatField('${pkt-dst-aws-service}');
+
+  /**
+   * The direction of the flow with respect to the interface where traffic is captured.
+   */
+  public static readonly FLOW_DIRECTION = new LogFormatField('${flow-direction}');
+
+  /**
+   * The path that egress traffic takes to the destination.
+   */
+  public static readonly TRAFFIC_PATH = new LogFormatField('${traffic-path}');
+
+  /**
+   * The default format.
+   */
+  public static readonly ALL_DEFAULT_FIELDS = new LogFormatField('${version} ${account-id} ${interface-id} ${srcaddr} ${dstaddr} ${srcport} ${dstport} ${protocol} ${packets} ${bytes} ${start} ${end} ${action} ${log-status} ${vpc-id} ${subnet-id} ${instance-id} ${tcp-flags} ${type} ${pkt-srcaddr} ${pkt-dstaddr} ${region} ${az-id} ${sublocation-type} ${sublocation-id} ${pkt-src-aws-service} ${pkt-dst-aws-service} ${flow-direction} ${traffic-path}');
```

Comment:
Edge case question here. Since this will be a manual update in the future, what happens if a customer uses this and then also adds a custom one because it's not in here yet, and then we update this to include it (so now it's a duplicate). Will that cause an error? Will it de-duplicate it? Or will it just have the field twice?

Reply:
In this implementation, the expanded string of each parameter is defined as a space-separated string. Thus there will be two fields.

Reply:
fixed.